src/libraries/System.Globalization/tests/CultureInfo/CultureInfoNames.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.RemoteExecutor;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoNames
+    {
+        private static bool IsIcuAndRemoteExecutionSupported => RemoteExecutor.IsSupported && PlatformDetection.IsIcuGlobalization;
+        [ConditionalTheory(nameof(IsIcuAndRemoteExecutionSupported))]
+        [InlineData("en", "en", "English", "English")]
+        [InlineData("en", "fr", "English", "anglais")]
+        [InlineData("aa", "aa", "Afar", "Afar")]
+        [InlineData("en-US", "en-US", "English (United States)", "English (United States)")]
+        [InlineData("en-US", "fr-FR", "English (United States)", "anglais (États-Unis)")]
+        [InlineData("en-US", "de-DE", "English (United States)", "Englisch (Vereinigte Staaten)")]
+        [InlineData("aa-ER", "aa-ER", "Afar (Eritrea)", "Afar (Eritrea)")]
+        [InlineData("", "en-US", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        [InlineData("", "fr-FR", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        [InlineData("", "", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        public void TestDisplayName(string cultureName, string uiCultureName, string nativeName, string displayName)
+        {
+            RemoteExecutor.Invoke((locale, uiLocale, native, display) =>
+            {
+                CultureInfo ci = CultureInfo.GetCultureInfo(locale);
+                CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo(uiLocale);
+
+                Assert.Equal(native, ci.NativeName);
+                Assert.Equal(display, ci.DisplayName);
+            }, cultureName, uiCultureName, nativeName, displayName).Dispose();
+        }
+
+        [ConditionalFact(nameof(IsIcuAndRemoteExecutionSupported))]
+        public void TestDisplayNameWithSettingUICultureMultipleTime()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                CultureInfo ci = CultureInfo.GetCultureInfo("en-US");
+                CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo("en-US");
+                Assert.Equal("English (United States)", ci.DisplayName);
+
+                CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo("fr-FR");
+                Assert.Equal("anglais (États-Unis)", ci.DisplayName);
+
+                CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo("de-DE");
+                Assert.Equal("Englisch (Vereinigte Staaten)", ci.DisplayName);
+            }).Dispose();
+        }
+    }
```

Comment:
Why do these tests need RemoteExecutor?

Reply:
I want to have Current UI culture always be set in isolated execution environment. This to void resetting it every time we use it. The other reason I want to restrict the test to the platforms support remote execution as ICU in such environments are trimmed and I expect the ICU there will not have all needed resources we depend on.

Reply:
I removed the remote execution from the tests and I'll the CI tell if this will succeed.

Reply:
> I want to have Current UI culture always be set in isolated execution environment. This to void resetting it every time we use it.

What's the problem with resetting it every time we use it?  We use the ThreadCultureChange helper in tons of tests:
https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrentInfo.cs#L25-L28
There's an issue with using it here?  To be clear, my concern is simply that the remote execution stuff adds considerable overhead, and we've culled back usage of it to only the cases where we really do need a clean slate where we fully control everything that gets run... I'm not clear why this falls into that category (if it does, though, ok).

Reply:
> I removed the remote execution from the tests and I'll let CI tell if this will succeed.

Ah, great, thanks!

Reply:
@stephentoub 
I am wondering about the behavior of the `ThreadCultureChange` if the execution code inside the using scope throw exception. does the dispose will get a chance to execute and restore the original cultures?